activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -134,7 +134,10 @@ def first!
     #
     #   Person.last # returns the last object fetched by SELECT * FROM people
     #   Person.where(["user_name = ?", user_name]).last
-    #   Person.order("created_on DESC").offset(5).last
+    #   Person.order("created_on DESC").offset(5).last # returns the last object (or nil if Person.all.size <= 5)
+    #
+    #  Note that the use of offset with last is deprecated, because relations with an offset cannot be automatically reversed.
```

Comment:
At the moment it's both, because `reverse_sql_order` will raise `IrreversibleOrderError`, which is already being caught and turned into a deprecation warning by the existing rescue block in `last`.

But it sounds from the comment below that you think going straight to `reverse_sql_order` raising an exception is too drastic a change, which is reasonable, so I can prepare a commit that will take care of that.